src/components/common/buttons/ShareButton.js

```diff
@@ -17,28 +19,19 @@ type ShareButtonProps = {
 }
 
 const log = logger.child({ from: 'ShareButton' })
+const SharingButton = isSharingAvailable ? CustomButton : CopyButton
 
-const ShareButton = ({ share, onPressDone, onPressed = noop, actionText, ...buttonProps }: ShareButtonProps) => {
-  const { canShare, shareAction } = useNativeSharing()
-  const shareUrl = isString(share) ? share : get(share, 'url', null)
+const ShareButton = ({ share, onPressed = noop, actionText, ...buttonProps }: ShareButtonProps) => {
+  const shareHandler = useNativeSharing(share, { onSharePress: onPressed })
 
   useEffect(() => {
     log.info('getPaymentLink', { share })
   }, [])
 
-  const shareHandler = useCallback(() => {
-    shareAction(share)
-    onPressed()
-  }, [shareAction, share])
-
-  return canShare ? (
-    <CustomButton onPress={shareHandler} {...buttonProps}>
-      {actionText}
-    </CustomButton>
-  ) : (
-    <CopyButton toCopy={shareUrl} onPress={onPressed} onPressDone={onPressDone} {...buttonProps}>
+  return (
+    <SharingButton {...buttonProps} toCopy={share} onPress={isSharingAvailable ? shareHandler : onPressed}>
```

Comment:
buttonProps should be at the end so they can override defaults

Reply:
i know it, that because i moved button props to the start - i won't the handlers / data to copy could be overridden
it's a built-in component functionality and should be incapsulated
the other button props of course are free to redefine